var/spack/repos/builtin/packages/dftbplus/package.py

```diff
@@ -0,0 +1,144 @@
+
+        And the last thing we do here is to set the installdir
+        """
+        dircwd = os.getcwd()
+        makefile = FileFilter("makefile")
+        makefile.filter("ROOT := .*", "ROOT := {0}".format(dircwd))
+
+        archmake = join_path(".", "sys", "make.x86_64-linux-gnu")
+        copy(archmake, join_path(dircwd, "make.arch"))
+
+        march = FileFilter(join_path(dircwd, 'make.arch'))
+
+        mconfig = FileFilter(join_path(dircwd, 'make.config'))
+
+        mconfig.filter('INSTALLDIR := .*', 'INSTALLDIR := {0}'.format(prefix))
+
+        if '+gpu' in self.spec:
+            march.filter('MAGMADIR = .*',
+                         'MAGMADIR = {0}'.format(spec['magma'].prefix))
+
+            mconfig.filter('WITH_GPU := .*', 'WITH_GPU := 1')
+
+        if '+mpi' in self.spec:
+            march.filter('SCALAPACKDIR = .*',
+                         'SCALAPACKDIR = {0}'.format(spec['scalapack'].prefix))
+
+            march.filter('LIB_LAPACK = -l.*',
+                         'LIB_LAPACK = {0}'.format(spec['blas'].libs.ld_flags))
+
+            march.filter('mpifort', '{0}'.format(spec['mpi'].mpifc))
+
+            mconfig.filter('WITH_MPI := .*', 'WITH_MPI := 1')
+
+            if '+elsi' in self.spec:
+                mconfig.filter('WITH_ELSI := .*', 'WITH_ELSI := 1')
+
+                has_pexsi = '+enable_pexsi' in spec['elsi']
+
+                mconfig.filter('WITH_PEXSI := .*', 'WITH_PEXSI := {0}'.format(
+                    '1' if has_pexsi is True else '0'
+                ))
+
+                march.filter("ELSIINCDIR .*", "ELSIINCDIR = {0}".format(
+                    spec['elsi'].prefix.include
+                ))
+
+                march.filter("ELSIDIR .*",
+                             "ELSIDIR = {0}".format(spec['elsi'].prefix))
+
+        else:
+            march.filter('LIB_LAPACK += -l.*', 'LIB_LAPACK += {0}'.format(
+                spec['blas'].libs.ld_flags))
+
+        if '+sockets' in self.spec:
+            mconfig.filter('WITH_SOCKETS := .*', 'WITH_SOCKETS := 1')
+
+        if '+transport' in self.spec:
+            mconfig.filter('WITH_TRANSPORT := .*', 'WITH_TRANSPORT := 1')
+
+        if '+arpack' in self.spec:
+            march.filter('ARPACK_LIBS = .*', 'ARPACK_LIBS = {0}'.format(
+                spec['arpack-ng'].libs.ld_flags
+            ))
+
+            mconfig.filter('WITH_ARPACK := .*', 'WITH_ARPACK := 1')
+
+        if '+dftd3' in self.spec:
+            march.filter('COMPILE_DFTD3 = .*', 'COMPILE_DFTD3 = 0')
+            march.filter('DFTD3_INCS = .*', 'DFTD3_INCS = -I{0}'.format(
+                spec['dftd3-lib'].prefix.include
+            ))
+
+            march.filter('DFTD3_LIBS = .*',
+                         'DFTD3_LIBS = -L{0} -ldftd3'.format(
+                             spec['dftd3-lib'].prefix))
+
+            mconfig.filter('WITH_DFTD3 := .*', 'WITH_DFTD3 := 1')
+
+        # Download some data needed for the tests
+        Executable('./utils/get_opt_externals')()
```

Comment:
This might be problematic. Spack is designed to work on air-gapped networks, so all downloads should be registered to Spack. I’ll have to take a closer look at this script, but you can probably replace it with some `resource` calls. There should be a section in the Packaging Guide about resources, but the docs aren’t great.

Reply:
Fixed in the latest commit.